### PR TITLE
feat: strict helmfile.yaml ext when render as template

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

feat: strict helmfile.yaml ext when render as a template

see: https://github.com/helmfile/helmfile/blob/main/docs/proposals/towards-1.0.md

key case: TestLoadDesiredStateFromYaml_EnvvalsInheritanceToBaseTemplate

I change the `loadFileWithOverrides` func for the `bases` feature.

### CODE REVIEW

**Review by itscaro — APPROVED** [review]

**Comment by mumoshu** [comment]
We'd need a ability to toggle this feature via a build flag or an envvar. Otheriwse we might need to maintain two branches, one for v1.0 containing this commit, another for v0.x. What's your preference? @itscaro

**Comment by yxxhero** [comment]
@yxxhero I think so.

File: pkg/app/desired_state_file_loader.go [hunk 1]
```diff
@@ -188,33 +173,40 @@ func (a *desiredStateLoader) load(yaml []byte, baseDir, file string, evaluateBas
...
-
 		id := fmt.Sprintf("%s.part.%d", filename, i)
 
-		if env == nil && overrodeEnv == nil {
-			yamlBuf, err = ld.renderTemplatesToYaml(baseDir, id, part)
-			if err != nil {
-				return nil, fmt.Errorf("error during %s parsing: %v", id, err)
+		var rawContent []byte
+
+		if isRequiredRender(filename) {
```
Comment (mumoshu) [inline pkg/app/desired_state_file_loader.go:188]: @yxxhero How is this function defined? (I see no definition in sources 🤔

**Comment by mumoshu** [comment]
@yxxhero Can we probably close this as done given we've already merged https://github.com/helmfile/helmfile/pull/599?
Please let me know if I missed anything!

**Comment by mumoshu** [comment]
@yxxhero Similarly to https://github.com/helmfile/helmfile/pull/442#issuecomment-1374353494, shall we turn this into a PR that is mergeable very soon, by adding another test files for v1for v0 tests you touched in this pull request instead of updating those in-place?
That way, we can maintain both v0 and v1 variants of the same tests for a while, and remove v0 tests after Helmfile v0.x finally reach EOL.
Otherwise, we can make every affected test to run sub-tests for v0 and v1 like I've done in https://github.com/helmfile/helmfile/blob/fdfa520f1ba80dbbe7764755b93c883e517973cf/pkg/app/app_test.go#L4414-L4422, and mark the v0 subest with a code comment like `TODO: Remove this once Helmfile v0.x reaches EOL` for easy removal.

**Comment by yxxhero** [comment]
@mumoshu I saw the PR again. I think that this PR can be closed. when helmfile v0.x reaches EOL. we just update the ext of test files, then all is ok.

**Comment by mumoshu** [comment]
@yxxhero That would also work as we already have some test coverage for the new file extension via https://github.com/helmfile/helmfile/pull/609. Thanks a lot for your confirmation ☺️

**Pull request closed without merging**